include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,978 @@
+/* Copyright 2017-2018 PaGMO development team
+ This file is part of the PaGMO library.
+ The PaGMO library is free software; you can redistribute it and/or modify
+ it under the terms of either:
+ * the GNU Lesser General Public License as published by the Free
+ Software Foundation; either version 3 of the License, or (at your
+ option) any later version.
+ or
+ * the GNU General Public License as published by the Free Software
+ Foundation; either version 3 of the License, or (at your option) any
+ later version.
+ or both in parallel, as here.
+ The PaGMO library is distributed in the hope that it will be useful, but
+ WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ for more details.
+ You should have received copies of the GNU General Public License and the
+ GNU Lesser General Public License along with the PaGMO library.  If not,
+ see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_ALGORITHMS_GACO_HPP
+#define PAGMO_ALGORITHMS_GACO_HPP
+
+#include <algorithm> // std::shuffle, std::transform
+#include <boost/math/constants/constants.hpp>
+#include <iomanip>
+#include <numeric> // std::iota, std::inner_product
+#include <random>
+#include <sstream> //std::osstringstream
+#include <string>
+#include <tuple>
+#include <utility> //std::swap
+#include <vector>
+
+#include <pagmo/algorithm.hpp> // needed for the cereal macro
+#include <pagmo/exceptions.hpp>
+#include <pagmo/io.hpp>
+#include <pagmo/population.hpp>
+#include <pagmo/problem.hpp>
+#include <pagmo/problems/decompose.hpp>
+#include <pagmo/rng.hpp>
+#include <pagmo/utils/constrained.hpp>
+#include <pagmo/utils/generic.hpp>
+#include <pagmo/utils/generic.hpp>         // uniform_real_from_range
+#include <pagmo/utils/multi_objective.hpp> // crowding_distance, etc..
+
+namespace pagmo
+{
+/// Extended ACO
+/**
+ * ACO is inspired by the natural mechanism with which real ant colonies forage food.
+ * This algorithm has shown promising results in many trajectory optimization problems.
+ * The first appearance of the algorithm happened in Dr. Marco Dorigo's thesis, in 1992.
+ * ACO generates future generations of ants by using the a multi-kernel gaussian distribution
+ * based on three parameters (i.e., pheromone values) which are computed depending on the quality
+ * of each previous solution. The solutions are ranked through an oracle penalty method.
+ *
+ *
+ * The version implemented in pagmo can be applied to box-bounded multiple-objective optimization and its implementation
+ * is an extension of Schlueter's originally proposed ACO algorithm.
```

Comment:
@darioizzo 
I am currently working in the documentation. I thought it could be nice to refer also to my MSc thesis, so that maybe who will be interested can also check it out. 
I was thus thinking to substitute line 60 with a note, in which I would write something like:

The ACO version implemented in PaGMO is an extension of Schlueter's originally proposed ACO algorithm and it was developed by Giacomo Acciarini in collaboration with the PaGMO development team, during his Aerospace Engineering MSc Thesis in TU Delft.

Would this be okay/appropriate?

Reply:
No, when the thesis is online you may do a PR adding the link to it, but for the time being it only creates a broken link (the thesis is not available).